src/categorical_algebra/Diagrams.jl

```diff
@@ -244,11 +266,68 @@ op(f::DiagramHom{op}) = DiagramHom{id}(op(shape_map(f)), op(diagram_map(f)),
 function compose(d::Diagram{T}, F::Functor; kw...) where T
   Diagram{T}(compose(diagram(d), F; kw...))
 end
-function compose(f::DiagramHom{T}, F::Functor; kw...) where T
-  DiagramHom{T}(shape_map(f), composeH(diagram_map(f), F; kw...),
+function compose(f::DiagramHom{T}, F::Functor; params = [], kw...) where T
+  whiskered = isempty(params) ? composeH(diagram_map(f), F; kw...) : param_compose(diagram_map(f),F;params=params)
+  DiagramHom{T}(shape_map(f), whiskered,
                 compose(f.precomposed_diagram, F; kw...))
 end
 
+"""Whisker a partially natural transformation with a functor ``H``,
+given any needed parameters specifying the functions in ``H``'s codomain
+which the whiskered result should map to. Currently assumes
+the result will be a totally defined transformation.
+"""
+
+Base.haskey(v::AbstractVector,i) = 1 <= i <= length(v)
+function param_compose(α::FinTransformation, H::Functor; params=[])
```

Comment:
Would be helpful to have a docstring for this function. I can't immediately tell what it's for besides it obviously having something to do with parameters.

Reply:
It was mere lines away!